Objects/exceptions.c

```diff
@@ -905,12 +905,22 @@ BaseExceptionGroup_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
         return NULL;
     }
 
-    exceptions = PySequence_Tuple(exceptions);
+    /* Make a copy of the passed exceptions sequence, for use in the repr */
+    PyObject *exceptions_copy = exceptions;
+    if (PyList_Check(exceptions)) {
+        exceptions_copy = PySequence_List(exceptions);
+    } else {
+        Py_INCREF(exceptions_copy);
+    }
```

Comment:
It would be better if we could do this without hurting performance of creating an ExceptionGroup. If we do a bit more work in repr() that's ok.